From: Marc Lainez <marc.lainez@gmail.com>
Date: Sun, 13 Jul 2026 11:30:00 +0200
Subject: [PATCH] jiffies.h: declare jiffies and jiffies_64 with
 ____cacheline_aligned_in_smp

GCC >= 10 promotes conflicting section attributes between declaration
(__jiffy_data -> .data) and definition (__cacheline_aligned_in_smp ->
.data..cacheline_aligned) to an error under -Werror=attributes.

Backport of mainline commit 7c30f352c852 ("jiffies.h: declare jiffies
and jiffies_64 with ____cacheline_aligned_in_smp").
---
--- a/include/linux/jiffies.h
+++ b/include/linux/jiffies.h
@@ -2,6 +2,7 @@
 #define _LINUX_JIFFIES_H
 
 #include <linux/math64.h>
+#include <linux/cache.h>
 #include <linux/kernel.h>
 #include <linux/types.h>
 #include <linux/time.h>
@@ -67,15 +68,17 @@
  * but that can only take up to, say, 4-byte variables. jiffies being part of
  * an 8-byte variable may not be correctly accessed unless we force the issue
  */
-#define __jiffy_data  __attribute__((section(".data")))
+#ifndef __jiffy_arch_data
+#define __jiffy_arch_data
+#endif
 
 /*
  * The 64-bit value is not atomic - you MUST NOT read it
  * without sampling the sequence number in jiffies_lock.
  * get_jiffies_64() will do this for you as appropriate.
  */
-extern u64 __jiffy_data jiffies_64;
-extern unsigned long volatile __jiffy_data jiffies;
+extern u64 __cacheline_aligned_in_smp jiffies_64;
+extern unsigned long volatile __cacheline_aligned_in_smp __jiffy_arch_data jiffies;
 
 #if (BITS_PER_LONG < 64)
 u64 get_jiffies_64(void);
